src/django/api/serializers.py

```diff
@@ -68,13 +73,83 @@ class Meta:
 
 class FacilitySerializer(GeoFeatureModelSerializer):
     oar_id = SerializerMethodField()
+    other_names = SerializerMethodField()
+    other_addresses = SerializerMethodField()
+    contributors = SerializerMethodField()
 
     class Meta:
         model = Facility
         fields = ('id', 'name', 'address', 'country_code', 'location',
-                  'created_at', 'updated_at', 'oar_id')
+                  'created_at', 'updated_at', 'oar_id', 'other_names',
+                  'other_addresses', 'contributors')
         geo_field = 'location'
 
     # Added to ensure including the OAR ID in the geojson properties map
     def get_oar_id(self, facility):
         return facility.id
+
+    def get_other_names(self, facility):
+        facility_list_item_matches = [
+            FacilityListItem.objects.get(pk=pk)
+            for (pk,)
+            in facility
+            .facilitymatch_set
+            .filter(status__in=[FacilityMatch.AUTOMATIC,
+                                FacilityMatch.CONFIRMED])
+            .values_list('facility_list_item')
+        ]
+
+        return {
+            match.name
+            for match
+            in facility_list_item_matches
+            if len(match.name) != 0
+            and match.name is not None
+            and match.name != facility.name
+            and match.facility_list.is_active
```

Comment:
This is some important logic. Since we are doing "soft deletes" we need to make sure that we are confident that we will never incorrectly show that a contributor is associated with a facility. It looks right, but I think we should consider adding some test coverage around it. If it is easier, we could move the bulk of the logic out to a method on the `FacilityListItem` model.

Reply:
Sounds good. I'll make these testable and add some unit tests around them.

Reply:
I moved these methods into the Facility model and then created a suite of unit tests in af190436461150f9473860f2f5b202204e489d00